[WIP] Improve Gatsby starter

AWolf81 opened the pull request:
**Related Issue:**
#47
 
**Summary:**
After merging #335 I've noticed that entering an url or a starter name wasn't available. The `Summary` mentioned it but the functionality was removed because of the last UX improvement. Summary is correct again as the features are available once this PR is merged.

**Added the following to the input field:**
- Select starter by name e.g. enter `gatsby-starter-blog`
- Select stater by url e.g. `https://github.com/gatsbyjs/gatsby-starter-blog`
- Allow custom url with-out selecting a starter from the list e.g. `https://github.com/2manyprojects2littletime/gatsby-starter-blog`
- Test cases for the added functionality

**Discussion**
- Deleting the entered term in the input won't deselect the starter. I think that's OK - just wanted to mention it.
- Do we need an info that the url entered is valid? At the moment, there is no indication that it will be used. That's probably OK. We're only displaying a message if the starter wasn't found. I think that's a rare use-case we can improve later.
- Search improvement - @melanieseltzer mentioned it but I don't know it yet. But we can improve it with this PR.

superhawk610 commented:
I agree with @melanieseltzer, it shouldn't autoselect. If it's going to autoselect, there should be a more clear indication that the highlighted result is selected (something like the text **SELECTED** next to the result).

+1 for the `No description` text, I also think it looks better for consistency.

AWolf81 commented:
@melanieseltzer I've added a `No description` text for `n/a` entries so it looks consistent.

For the selection I've added a text to display that it will be selected. Styling is a bit basic and could probably be improved.

I think it's OK to autoselect as most user will probably just select a starter from the list and won't type it in the input box. They're typing it if they want to use a starter that's not listed in the starter list.

melanieseltzer commented on src/components/CreateNewProjectWizard/Gatsby/SelectStarterList.js:
```diff
@@ -131,6 +136,13 @@ const Description = styled.p`
+  background: ${COLORS.gray[200]};
+  border-radius: 5px;
+  padding: 5px;
+  display: ${props => (props.visible ? 'auto' : 'none')};
```
`auto` isn't valid for display :-) We can conditionally set the whole property since `display: block` is the default:

```js
${({ visible }) => !visible && `
  display: none;
`}
```

AWolf81 replied:
ups, sorry, you're right. 👍 I added the code as you proposed
